### PR TITLE
WLAN beacon implementation

### PR DESCRIPTION
**Pull request by BluemarkInnovations**
First PR of the WLAN Beacon implementation.

- ~~Beacon functionality only works if also WLAN NaN is enabled (and the web interface). Needs more testing, it seems that without WLAN NaN, the module uses channel 1 and the PR doesn't work.~~
- ~~Need improvement, so the module transmits also WLAN Beacon packets if the web interface is disabled.~~
- The Beacon update rate means that the RID information is updated with this rate. Default an AP transmits every 100 ms a Beacon packet.
- Added the probe response implementation. This is the default mechanism (probe request/response) for smartphones to detect nearby APs. Without the probe-response implementation, the user experience/update rate is horrible on most phones. With the probe response implementation and on the smartphone, WiFi scanning throttling off (Developer options, see the OpenDroneID project), performance is acceptable.

FYI @friissoren

### CODE REVIEW

**Comment by BluemarkInnovations** [comment]
Fixed outstanding issues

File: RemoteIDModule/WiFi_TX.cpp [hunk 1]
```diff
@@ -21,31 +21,38 @@ bool WiFi_NAN::init(void)
...
 
-    wifi_config.ap.ssid_hidden = 1;
-    wifi_config.ap.channel = wifi_channel;
+    wifi_config.ap.ssid_hidden = 0; //make always visible
+    wifi_config.ap.channel = g.wifi_channel;
 
     if (esp_wifi_set_config(WIFI_IF_AP, &wifi_config) != ESP_OK) {
         return false;
-    }
+    }*/
```
Comment (friissoren) [inline RemoteIDModule/WiFi_TX.cpp:41]: Why is this commented out? If the code is not needed, it is probably better to delete it.

File: RemoteIDModule/WiFi_TX.cpp [hunk 2]
```diff
@@ -21,31 +21,38 @@ bool WiFi_NAN::init(void)
...
     if (esp_wifi_set_bandwidth(WIFI_IF_AP, WIFI_BW_HT20) != ESP_OK) {
         return false;
     }
-    if (esp_read_mac(WiFi_mac_addr, ESP_MAC_WIFI_STA) != ESP_OK) {
+    /*if (esp_read_mac(WiFi_mac_addr, ESP_MAC_WIFI_STA) != ESP_OK) {
         return false;
-    }
+    }*/
```
Comment (friissoren) [inline RemoteIDModule/WiFi_TX.cpp:47]: Same here. Why is this commented out?

File: RemoteIDModule/WiFi_TX.cpp (hunk 1 above)
Comment (BluemarkInnovations) [reply RemoteIDModule/WiFi_TX.cpp]: well I noticed after the commit I needed to clean up a bit, I have done that now.

File: RemoteIDModule/WiFi_TX.cpp (hunk 2 above)
Comment (BluemarkInnovations) [reply RemoteIDModule/WiFi_TX.cpp]: well I noticed after the commit I needed to clean up a bit, I have done that now.

**Comment by BluemarkInnovations** [comment]
- made the wifi channel a parameter
- code clean up, some calls were not needed
- the issue/bug was that the web server did have its own default configuration values, not using the configured ones, solved in this PR
- use the random generated MAC address also for WLAN packet generation instead of the device MAC address
- added separate counters for both beacon and NaN packets

**Comment by tridge** [comment]
squashed commits

**Pull request merged**